.github/workflows/update-cmake.yml

```diff
@@ -0,0 +1,55 @@
+name: Update CMake
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/update-cmake.yml'
+      - 'scripts/update_cmake_version.py'
+      - 'CMakeUrls.cmake'
+      - 'docs/index.rst'
+      - 'docs/update_cmake_version.rst'
+      - 'README.rst'
+      - 'tests/test_distribution.py'
+      - 'noxfile.py'
+  workflow_dispatch:
+  schedule:
+    - cron: '0 6 * * *'  # "At 06:00 every day."
+
+jobs:
+  update-dependencies:
+    name: Update dependencies
+    if: github.repository_owner == 'scikit-build' || github.event_name != 'schedule'
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: wntrblm/nox@2022.11.21
+    - name: "Run update: bump CMake"
+      id: bump-cmake
+      run: |
+        nox --force-color -s bump
+        echo "version=$(cat .cmake.version)" >> $GITHUB_OUTPUT
+
+    # we use this step to grab a Github App auth token, so that PRs generated by this workflow
+    # run the GHA tests.
+    - uses: tibdex/github-app-token@v1
+      id: generate-token
+      if: github.ref == 'refs/heads/master' && github.repository == 'scikit-build/cmake-python-distributions'
+      with:
+        app_id: ${{ secrets.SCIKIT_BUILD_BOT_APP_ID }}
```

Comment:
a bot needs to be setup to get a proper token.

Reply:
Do you need this if you select the "create PRs" permission for GHA?

Reply:
from: https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#workarounds-to-trigger-further-workflow-runs
I understand that a PR could be created using the default GITHUB_TOKEN, however, this PR would not trigger a CI workflow run.

Reply:
I created an app at the organization level as users are less appropriate for this.

The app is installed only for this repo.
Secrets are set at organization level but limited to this repo for now.